symengine/dense_matrix.cpp

```diff
@@ -327,19 +327,27 @@ void add_dense_scalar(const DenseMatrix &A, const RCP<const Basic> &k,
 // ------------------------------- Matrix Multiplication ---------------------//
 void mul_dense_dense(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
 {
-    SYMENGINE_ASSERT(A.col_ == B.row_ and C.row_ == A.row_
-                     and C.col_ == B.col_);
+    SYMENGINE_ASSERT(A.col_ == B.row_);
+
+    if (C.row_ != A.row_ or C.col_ != B.col_)
+        C.resize(A.row_, B.col_);
```

Comment:
Since we don't have a clear consensus, let's remove the changes above and keep the changes below